notation.go

```diff
@@ -188,21 +199,23 @@ func Sign(ctx context.Context, signer Signer, repo registry.Repository, signOpts
 	logger.Debug("Generating annotation")
 	annotations, err := generateAnnotations(signerInfo, pluginAnnotations)
 	if err != nil {
-		return ocispec.Descriptor{}, err
+		return ocispec.Descriptor{}, ocispec.Descriptor{}, err
 	}
 	logger.Debugf("Generated annotations: %+v", annotations)
-	logger.Debugf("Pushing signature of artifact descriptor: %+v, signature media type: %v", targetDesc, signOpts.SignatureMediaType)
-	_, _, err = repo.PushSignature(ctx, signOpts.SignatureMediaType, sig, targetDesc, annotations)
+	logger.Debugf("Pushing signature of artifact descriptor: %+v, signature media type: %v", artifactManifestDesc, signOpts.SignatureMediaType)
+	_, sigManifestDesc, err = repo.PushSignature(ctx, signOpts.SignatureMediaType, sig, artifactManifestDesc, annotations)
 	if err != nil {
 		var referrerError *remote.ReferrersError
-
-		// do not log an error for failing to delete referral index
-		if !errors.As(err, &referrerError) || !referrerError.IsReferrersIndexDelete() {
-			logger.Error("Failed to push the signature")
+		if errors.As(err, &referrerError) && referrerError.IsReferrersIndexDelete() {
+			// log warning if referrers index removal failed but signature
+			// pushed succeeded
+			logger.Warn("Removal of outdated referrers index from remote registry failed. Garbage collection may be required.")
+			return artifactManifestDesc, sigManifestDesc, nil
```

Comment:
What's this change for? If it is not related to `SignOCI`, please propose this change in another PR.

Reply:
This change is required for outputting the signature manifest digest because when we return a referrers index removal failed error, the `artifactManifestDesc` and `sigManifestDesc` are empty values, and we cannot output the signature manifest digest, but this a success case. Previously, we didn't use the returned descriptor, so this wasn't an issue. 
You can look at the Notation PR to understand what I'm saying. https://github.com/notaryproject/notation/pull/1204

Reply:
It changes the behavior.

Reply:
Updated to return the error.